### PR TITLE
WRR-12055: Fixed typos and JSDocs

### PR DESCRIPTION
**Pull request by ion-andrusciac-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] I have run automated CLI testing and it is passed
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Fixed typos and JSDocs

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
- Fixed typos and grammar errors
- Removed unused function arguments and imports 
- Fixed JSDocs

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRR-12055

### Comments

Enact-DCO-1.0-Signed-off-by: Ion Andrusciac ion.andrusciac@lgepartner.com

### CODE REVIEW

File: lib/rules/prop-types.js [hunk 1]
```diff
@@ -69,8 +69,8 @@ module.exports = {
 
     /**
      * Helper for accessing the current scope in the stack.
-     * @param {string} key The name of the identifier to access. If omitted, returns the full scope.
-     * @param {ASTNode} value If provided sets the new value for the identifier.
+     * @param {string} [key] The name of the identifier to access. If omitted, returns the full scope.
+     * @param {ASTNode} [value] If provided sets the new value for the identifier.
```
Comment (daniel-stoian-lgp) [inline lib/rules/prop-types.js:73]: why was this change necessary?

File: README.md [hunk 2]
```diff
@@ -79,7 +79,7 @@ Finally, enable all of the rules that you would like to use:
 
 # List of supported rules
 
-* [enact/kind-name](docs/rules/kind-name.md): Prevent missing `name` in a Enact component definition
+* [enact/kind-name](docs/rules/kind-name.md): Prevent missing `name` in an Enact component definition
```
Comment (seunghoh) [inline README.md:82]: sharp eyes 👀

File: lib/rules/prop-types.js (hunk 1 above)
Comment (ion-andrusciac-lgp) [reply lib/rules/prop-types.js:73]: because the parameters are optional

**Review by daniel-stoian-lgp — APPROVED** [review]
LGTM

File: lib/rules/kind-name.js [hunk 3]
```diff
@@ -53,13 +53,7 @@ module.exports = {
...
         case 'PropertyDefinition':
           var tokens = sourceCode.getFirstTokens(node, 2);
-          if (
-            tokens[0].value === 'name' ||
-            (tokens[1] && tokens[1].value === 'name')
-          ) {
-            return true;
-          }
-          return false;
+          return !!(tokens[0].value === 'name' || (tokens[1] && tokens[1].value === 'name'));
```
Comment (seunghoh) [inline lib/rules/kind-name.js:56]: How about keeping the original code? The logic is the same but for better readablility.

File: lib/rules/kind-name.js [hunk 4]
```diff
@@ -133,14 +127,7 @@ module.exports = {
...
 
-      if (
-        namedObjectAssignment || namedObjectDeclaration ||
-        namedClass ||
-        namedFunctionDeclaration || namedFunctionExpression
-      ) {
-        return true;
-      }
-      return false;
+      return !!(namedObjectAssignment || namedObjectDeclaration || namedClass || namedFunctionDeclaration || namedFunctionExpression);
```
Comment (seunghoh) [inline lib/rules/kind-name.js:130]: How about keeping the original code? The logic is the same but for better readability.

File: lib/rules/display-name.js [hunk 5]
```diff
@@ -49,13 +49,7 @@ module.exports = {
...
         case 'PropertyDefinition':
           var tokens = sourceCode.getFirstTokens(node, 2);
-          if (
-            tokens[0].value === 'displayName' ||
-            (tokens[1] && tokens[1].value === 'displayName')
-          ) {
-            return true;
-          }
-          return false;
+          return !!(tokens[0].value === 'displayName' || (tokens[1] && tokens[1].value === 'displayName'));
```
Comment (seunghoh) [inline lib/rules/display-name.js:52]: Same here, let's keep the original code for better readability.

File: lib/rules/display-name.js [hunk 6]
```diff
@@ -130,14 +124,7 @@ module.exports = {
...
 
-      if (
-        namedObjectAssignment || namedObjectDeclaration ||
-        namedClass ||
-        namedFunctionDeclaration || namedFunctionExpression
-      ) {
-        return true;
-      }
-      return false;
+      return !!(namedObjectAssignment || namedObjectDeclaration || namedClass || namedFunctionDeclaration || namedFunctionExpression);
```
Comment (seunghoh) [inline lib/rules/display-name.js:127]: Same here, let's keep the original code for better readability.

File: lib/rules/prop-types.js [hunk 7]
```diff
@@ -171,13 +171,7 @@ module.exports = {
...
       if (node && (node.type === 'ClassProperty' || node.type === 'PropertyDefinition')) {
         var tokens = context.getFirstTokens(node, 2);
-        if (
-          tokens[0].value === 'propTypes' ||
-          (tokens[1] && tokens[1].value === 'propTypes')
-        ) {
-          return true;
-        }
-        return false;
+        return !!(tokens[0].value === 'propTypes' || (tokens[1] && tokens[1].value === 'propTypes'));
```
Comment (seunghoh) [inline lib/rules/prop-types.js:174]: Same here, let's keep the original.

File: lib/rules/prop-types.js [hunk 8]
```diff
@@ -674,7 +669,6 @@ module.exports = {
                 scanUsage(node);
               } else if (isComputed && prop.key && prop.key.name === 'styler') {
                 // Ignore 'styler' inside computed functions
-                continue;
```
Comment (seunghoh) [inline lib/rules/prop-types.js:677]: Would you explain why you removed this line?

File: lib/util/Components.js [hunk 9]
```diff
@@ -79,7 +79,7 @@ Components.prototype.set = function(node, props) {
  *
  * @returns {Object} Components list
  */
-Components.prototype.list = function(filter) {
+Components.prototype.list = function() {
```
Comment (seunghoh) [inline lib/util/Components.js:82]: `filter`is used in the function. Are you sure to be OK if we remove?

**Review by seunghoh — CHANGES_REQUESTED** [review]

File: lib/util/Components.js (hunk 9 above)
Comment (ion-andrusciac-lgp) [reply lib/util/Components.js:82]: In this case, the `filter` from the function parameter is not used. The used `filter` is the method from the `Array` instance

**Review by seunghoh — APPROVED** [review]
